### ISSUE TITLE
As drag placeholder render, dom cannot trigger button click event which in grid item at v1.4.3

### ISSUE THREAD

**Issue by moonrailgun** [react-grid-layout/react-grid-layout#1980]
### Describe the bug

I wanna add a button ( for example delete this item) in grid item.

And in latest version, its not work. Because its will trigger onDrag event and render a placeholder on click event trigger

Relation PR: https://github.com/react-grid-layout/react-grid-layout/pull/1923

### Your Example Website or App

https://codesandbox.io/s/vigorous-cdn-ssnvts?file=/src/ShowcaseLayout.js

### Steps to Reproduce the Bug or Issue

1. open sandbox
2. click "Click" button in grid item, its will not print log, onClick event is not work
3. switch package version from 'v1.4.3' to 'v1.4.2'
4. try again of step 2, every thing is work

### Expected behavior

As a user, i wanna we can click button in grid item. not just drag

### react-grid-layout library version

v1.4.3

### Operating System Version

windows

### Browser

Chrome

### Screenshots or Videos

https://github.com/react-grid-layout/react-grid-layout/assets/6964737/4d53d272-91ad-4cbe-8bed-6f0442e6c311

**Comment by xams-creator** [comment]
我也遇到了同样的问题，尝试修改这段css代码
```
.react-grid-item.react-grid-placeholder {
  background: red;
  opacity: 0.2;
  transition-duration: 100ms;
  z-index: 2;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -o-user-select: none;
  user-select: none;
  pointer-events: none;
}
```
**` pointer-events: none;`**

**Comment by tzimmermann** [comment]
I think you should add the `draggableCancel` prop and put this className also on the button, I've [modified your example to work](https://codesandbox.io/s/wispy-brook-dmk5sy).

**Comment by moonrailgun** [comment]
> I think you should add the `draggableCancel` prop and put this className also on the button, I've [modified your example to work](https://codesandbox.io/s/wispy-brook-dmk5sy).

Its a good idea, can as be a workaround solution. but when click major block for grid item, its will show a red block, and when i move it, this will disappear. Its looks weird

**Comment by jonashirata99** [comment]
Had a similar problem. In my case, I have a onClick in the grid item itself to set it as active, so I cannot use the draggableCancel prop on that

**Comment by Thomas101** [comment]
This can also be an issue if you want the top-level item to be draggable and clickable.

We have a case where the draggable item can be clicked on for some interaction, or also just dragged. In `1.4.2` this works fine, however in `1.4.3` the onClick handler fails to fire.

**Comment by mattwatts-aimltd-uk** [comment]
Same problem here, reverting to 1.4.2 fixes the issue for me.

**Comment by magyarp** [comment]
@moonrailgun Just add onMouseDown={(event) => event.stopPropagation()} to the button and your problem is gone.

**Comment by JesperCaspeco** [comment]
I had the same problem, if I add `onMouseDown={(event) => event.stopPropagation()}` it fixes the issue.

**Comment by leonbloy** [comment]
Same problem here. 

Because we have lots of widgets, with different interactive components, to place `onMouseDown={(event) => event.stopPropagation()}` everywhere is rather cumbersome. 

This worked ok in 1.4.2 ?

Is this change of behaviour to be expected? I don't see any release notes or changelog.

**Comment by moonrailgun** [comment]
@leonbloy i already leave relation PR and codesandbox then you can check it by your self

**Comment by ddrapeau-jmdnsolutions** [comment]
I agree with @leonbloy that it is cumbersome. What if I have a bars chart that I need to click on any bar. Do I have to write onMouseDown=... on every bar? That does not make sense. Especially since it was working properly with v1.4.2, for me at least, it seems to be a bug needed to  be solved. If only I was skilled enough to help, I would fix it myself and share with the community. Unfortunately, I am not that skilled.

**Comment by magyarp** [comment]
@leonbloy @ddrapeau-jmdnsolutions Add z-index: 10 to your direct child div of GridLayout.

**Comment by wozien** [comment]
some problem. mark

**Comment by hth520** [comment]
> 如果您希望顶级项目可拖动和可单击，这也可能是一个问题。
> 
> 我们有一种情况，可以单击可拖动项目进行某些交互，或者也可以只是拖动。在`1.4.2`此工作正常，但是在`1.4.3`onClick 处理程序中无法触发。

Thank you for solving my problem

**Comment by latourna** [comment]
Based on @xams-creator and @magyarp suggestions, these two CSS only solutions work:
```
.react-grid-item.react-grid-placeholder {
  pointer-events: none;
}
```
```
.react-grid-item {
  z-index: 3;
}
```
However the end result is slightly different: with the first solution the placeholder will appear on mouse down but not with the second one.

**Comment by TeoAvignon** [comment]
@latourna Cheers a lot! Looks like it also fixes a bug I have where the placeholder get stuck in front of the div!

**Comment by huangnan20030709** [comment]
thank u

**Comment by huangnan20030709** [comment]
> Based on [@xams-creator](https://github.com/xams-creator) and [@magyarp](https://github.com/magyarp) suggestions, these two CSS only solutions work:
> 
> ```
> .react-grid-item.react-grid-placeholder {
>   pointer-events: none;
> }
> ```
> 
> ```
> .react-grid-item {
>   z-index: 3;
> }
> ```
> 
> However the end result is slightly different: with the first solution the placeholder will appear on mouse down but not with the second one.

thank u ！！！

**Comment by mcmxcdev** [comment]
What worked for me is the following:

```
.react-grid-item.react-grid-placeholder {
  z-index: 1;
}

.react-grid-item {
  z-index: 2;
}

.react-grid-item.react-draggable-dragging {
  z-index: 3;
}
```

**Comment by rawbin-** [comment]
I am using 1.5.2 and get the same problem. when I click on the react-grid-item it turns out a red background and disappear, and also I need to process the clickable elements in all place with the . draggableCancel prop.
Is there any common solution to enable this feature by default as it was OK in 1.4.2?

**Comment by rawbin-** [comment]
> What worked for me is the following:
> 
> ```
> .react-grid-item.react-grid-placeholder {
>   z-index: 1;
> }
> 
> .react-grid-item {
>   z-index: 2;
> }
> 
> .react-grid-item.react-draggable-dragging,
> .react-grid-item.resizing {
>   z-index: 3;
> }
> ```

works for me to solve the placeholder jumps up when click on the draggable area

**Comment by cyuan01** [comment]
I think the root issue though from the [change](https://github.com/react-grid-layout/react-grid-layout/pull/1923) in 1.4.3 that introduced this bug is that we are now setting `dragActive` to true `onDragStart` rather than just `onDrag`. I'm guessing that `onDragStart` being the "start of the drag" fires a bit more eagerly for lack of better words whereas the `onDrag` has a bit more of a delay or longer wait time on mouse events to really determine if the event is a drag or just like a mouse down click event. 

I guess `onDragStart`  _is_ the correct place to set `dragActive` to true but maybe it should have more logic to confirm the drag has truly started to set `dragActive` to true. Or as suggested in the PR, fix the original issue the PR intended to resolve with an alternative approach.

**Comment by STRML** [comment]
This is fixed in v2.2.0! The new `dragConfig.threshold` option (default 3px) prevents drag/placeholder from appearing on simple clicks.

Clicks on buttons and interactive elements now work correctly:
- Click without movement → no drag, click passes through
- Click + move 3px+ → drag starts

```jsx
// v2 API - clicks work, drags require movement
<GridLayout dragConfig={{ threshold: 3 }} ... />
```

The workarounds (`draggableCancel`, `pointer-events: none`) are no longer needed for basic click handling.

Please upgrade to v2.2.0+.
